src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -45,6 +60,23 @@ public SecretManagerIPCAgent(SecretManager secretManager, AuthorizationHandler a
         this.authorizationHandler = authorizationHandler;
     }
 
+    /**
+     * Use secret manager config to set the cloud call thread pool size. If not configured, default value is used.
+     *
+     * @param config component configuration
+     */
+    public void setCloudCallThreadPool(Topics config) {
+        int cloudCallQueueSize = Coerce.toInt(config.lookupTopics(CONFIGURATION_CONFIG_KEY, PERFORMANCE_TOPIC)
+                .findOrDefault(DEFAULT_CLOUD_REQUEST_SIZE, CLOUD_REQUEST_QUEUE_SIZE_TOPIC));
+        if (cloudCallQueueSize <= 0) {
+            logger.atWarn().kv("queueSize", DEFAULT_CLOUD_REQUEST_SIZE)
+                    .log("Using default value for cloud request " + "queue size as the configured value is invalid");
+            cloudCallQueueSize = DEFAULT_CLOUD_REQUEST_SIZE;
+        }
+        cloudCallThreadPool =
```

Comment:
need to handle shutdown of existing pool, the one from L49 is still alive at this point